test/realtime_web/controllers/tenant_controller_test.exs

```diff
@@ -127,16 +131,36 @@ defmodule RealtimeWeb.TenantControllerTest do
 
     setup %{tenant: tenant} do
       [extension] = tenant.extensions
-      args = Map.put(extension.settings, "id", random_string())
-      start_supervised!({Extensions.PostgresCdcStream.Replication, args})
+      args = Map.put(extension.settings, "id", tenant.external_id)
+      {:ok, _} = Realtime.PostgresCdc.connect(Extensions.PostgresCdcStream, args)
+      on_exit(fn -> Realtime.PostgresCdc.stop_all(tenant) end)
       :ok
     end
 
     test "deletes chosen tenant", %{conn: conn, tenant: tenant} do
       with_mock JwtVerification, verify: fn _token, _secret -> {:ok, %{}} end do
+        {:ok, db_conn} = Helpers.check_tenant_connection(tenant, "realtime_test")
+
+        assert %{rows: [["supabase_realtime_replication_slot"]]} =
+                 Postgrex.query!(db_conn, "SELECT slot_name FROM pg_replication_slots", [])
+
         conn = delete(conn, Routes.tenant_path(conn, :delete, tenant.external_id))
-        :timer.sleep(10000)
         assert response(conn, 204)
+        :timer.sleep(1000)
+
+        {:ok, db_conn} =
+          start_supervised(
+            {Postgrex,
+             [
+               host: "localhost",
+               database: "realtime_test",
+               username: "postgres",
+               password: "postgres",
+               after_connect: &Postgrex.query!(&1, "set search_path=realtime", [])
+             ]}
+          )
```

Comment:
Interesting! Why are you starting this new connection here?

Reply:
The other way to start a connection depends on the tenant which fails after tenant deletion.